cmd/kube-bench-summarizer/summarizer/summarizer.go

```diff
@@ -0,0 +1,200 @@
+package summarizer
+
+import (
+	"github.com/sirupsen/logrus"
+	"io/ioutil"
+
+	"encoding/json"
+	"fmt"
+	kb "github.com/aquasecurity/kube-bench/check"
+	"os/exec"
+	"path/filepath"
+)
+
+const (
+	DEFAULT_OUTPUT_FILE_NAME = "report.json"
+)
+
+type Summarizer struct {
+	InputDirectory   string
+	OutputDirectory  string
+	summarizedReport *SummarizedReport
+	groupsMap        map[string]*GroupWrapper
+	checksMaps       map[string]*CheckWrapper
+}
+
+type CheckWrapper struct {
+	ID          string                 `yaml:"id" json:"test_number"`
+	Text        string                 `json:"test_desc"`
+	Audit       string                 `json:"-"`
+	Type        string                 `json:"-"`
+	Commands    []*exec.Cmd            `json:"-"`
+	Tests       map[string]interface{} `json:"-"`
+	Set         bool                   `json:"-"`
+	Remediation string                 `json:"-"`
+	TestInfo    []string               `json:"test_info"`
+	State       kb.State               `json:"status"`
+	ActualValue string                 `json:"-"`
+	Scored      bool                   `json:"-"`
+	Hosts       []string               `json:"hosts"`
+}
+
+type GroupWrapper struct {
+	ID     string          `yaml:"id" json:"section"`
+	Pass   int             `json:"pass"`
+	Fail   int             `json:"fail"`
+	Warn   int             `json:"warn"`
+	Info   int             `json:"info"`
+	Text   string          `json:"desc"`
+	Checks []*CheckWrapper `json:"results"`
+}
+
+type SummarizedReport struct {
+	Version string          `json:"-"`
+	Groups  []*GroupWrapper `json:"tests"`
+	kb.Summary
+}
+
+func NewSummarizer(inputDir, outputDir string) *Summarizer {
+	return &Summarizer{
+		InputDirectory:  inputDir,
+		OutputDirectory: outputDir,
+		summarizedReport: &SummarizedReport{
+			Groups: []*GroupWrapper{},
+		},
+		groupsMap:  map[string]*GroupWrapper{},
+		checksMaps: map[string]*CheckWrapper{},
+	}
+}
+
+func (s *Summarizer) summarizeForHost(hostname string) error {
+	logrus.Debugf("summarizeForHost: %v", hostname)
+
+	hostDir := fmt.Sprintf("%v/%v", s.InputDirectory, hostname)
+	resultFiles, err := filepath.Glob(fmt.Sprintf("%v/*.json", hostDir))
+	if err != nil {
+		return err
+	}
+
+	for _, fileName := range resultFiles {
+		// Load one result file
```

Comment:
Nit: it's better for the method to just return the error, and get it logged in the caller stack.

Reply:
So one method would be for readFile, another for readGroup